File: output/CodeGen/avatar/Java/Python/codeforces_49_A.py
def main():
    sc = Scanner(sys.stdin)
    s = sc.next()
    length = len(s)
    ch = 0
    for i in range(length-1,-1,-1):
        if s[i]!='' and s[i]!= '?':
            ch = s[i]
            break
    ch = ch.lower()
    if ch == 'a' or ch == 'e' or ch == 'i' or ch == 'o' or ch == 'u' or ch == 'y':
        print("YES")
    else:
        print("NO")

if __name__ == "__main__":
    main()

File: output/CodeGen/avatar/Python/Java_repair/codeforces_49_A.java
import java.util.*;

public class codeforces_49_A {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        String a = in.next();
        String[] vo = {"a","e","i","o","u","y","A","E","I","O","U","Y"};
        for(int i=0;i<a.length()-1;i++){
            if(a.charAt(i)==''|| a.charAt(i)=='?') continue;
            else if(a.charAt(i) in vo) System.out.println("YES");
            else System.out.println("NO");
            break;
        }
    }
}